CRM/Core/ManagedEntities.php

```diff
@@ -135,10 +135,15 @@ private function backfillDefaults(array &$item): void {
    * Take appropriate action on every managed entity.
    *
    * @param array[] $plan
+   * @param bool $force
+   *   If TRUE, then unconditionally re-save all entities. Use this if you suspect a data-integrity problem.
+   *   If FALSE, then only re-save entities when the declaration changes.
    */
-  private function reconcileEntities(array $plan): void {
+  private function reconcileEntities(array $plan, bool $force = FALSE): void {
```

Comment:
Non-blocker:

This new `$force` param looks to be deeply internal; it's not even reachable when calling the `public function reconcile()`. If it's useful, then maybe it should be exposed there and also as a param in the `Api4.Managed.reconcile` action.
Or if it's not useful, then maybe it should be removed per YAGNI?

Reply:
This was one of the questions raised in the description.

After looking at the tests, it seemed apparent that we need to re-save whenever an extension is toggled (i.e. propagating `is_active`). 

I also think that doing it during a system-upgrade (but not a manual flush) strikes a balance between (a) not needing to re-save all time and (b) still giving some enforcement to `update=>always`.

Anyway, the parameter is replaced by a computation now.